### DISCUSSION TITLE
How do I do a implementation on a method which returns bytes?

### DISCUSSION THREAD

**Discussion by KCD3V** [vfsfitvnm/frida-il2cpp-bridge#408] [Q&A]
I have the following method I'd like to implement:
```
public static byte[] ToByteArray(this IMessage message)
{
	return null;
}
```
And here is how I've set it up:
```
import "frida-il2cpp-bridge";
Il2Cpp.perform(() => {
    const GoogleProtos = Il2Cpp.domain.assembly("Google.Protobuf").image;
    const MagicClass = GoogleProtos.class("Google.Protobuf.Magic");
    MagicClass.method("ToByteArray").implementation = function(value){
        console.log('What is the value?', value);
        this.method('ToByteArray').invoke(value);
    }
});
```
Keep getting a crash, I assume I need to be casting value to <byte> but I've tried all sorts.

**Comment by fortiZde** [comment]
> I have the following method I'd like to implement:
> 
> ```
> 
> public static byte[] ToByteArray(this IMessage message)
> 
> {
> 
> 	return null;
> 
> }
> 
> ```
> 
> And here is how I've set it up:
> 
> ```
> 
> import "frida-il2cpp-bridge";
> 
> Il2Cpp.perform(() => {
> 
>     const GoogleProtos = Il2Cpp.domain.assembly("Google.Protobuf").image;
> 
>     const MagicClass = GoogleProtos.class("Google.Protobuf.Magic");
> 
>     MagicClass.method("ToByteArray").implementation = function(value){
> 
>         console.log('What is the value?', value);
> 
>         this.method('ToByteArray').invoke(value);
> 
>     }
> 
> });
> 
> ```
> 
> Keep getting a crash, I assume I need to be casting value to <byte> but I've tried all sorts.
> 
> 
> 
> 
> 
> 

Did you solve it?

**Reply by KCD3V** [reply]
```
const ByteValue = value as Il2Cpp.Object
        console.log('value', this.method('ToByteArray').invoke(ByteValue))
```

Yes thank you, not used to TypeScript is all
